### DISCUSSION TITLE
surface boundary are not correct even after using the transform method

### DISCUSSION THREAD

**Discussion by Marwa-AL** [cityjson/specs#164] [Q&A] [answered]
Hi, 

I am working with LOD2 data and I would like to extract surface boundaries of the ground floor. The second coordinate is out of the geographical extent even after I applied transform as mentioned in already existing question. Is there anything that I'm missing in this regard. 

here is the bbox: [697966.951, 5329984.909, 531.194, 700005.83, 5332068.84, 578.771]

and this's the boundary of the GroundSurface after transform: [(698665.143621, 5338498.643849, 531.7301799999999), (698665.145342, 5338498.649761, 531.7301799999999), (698665.145554, 5338498.65049, 531.7301799999999), (698665.151849, 5338498.648655, 531.7301799999999), (698665.148299, 5338498.636515, 531.7301799999999), (698665.142013, 5338498.638351, 531.7301799999999)]

**Comment by kenohori** [comment]
It's not really possible to say anything without looking at the data you're using. Maybe the geographical extent is incorrect?

**Comment by Marwa-AL** [comment]
Thank you for your reply! The geographical extent is the boundary box I got using .get_bbox(). It's also identical to the one I see trying to view the citygml file with FZKViewer. So, I believe that's right. 
I am using lod2 data downloaded from BayrenAtlas as CityGML and converted to CityJSON.

**Comment by hugoledoux** [comment, marked as answer]
`.get_bbox()` is a function of citygml4j I assume here? Then I'd say perhaps there is bug there?

You can try to change the CRS with cjio (operator "crs_reproject") to see what is wrong too. This recalculates the bbox.

**Reply by Marwa-AL** [reply]
I think the issue is not in the boundary box. It could be that I'm missing something in the transformation which I cannot identify.
example error is that area: True area is 118.93 --> calculated area from polygon formed with the vertices is 0.00011893

**Reply by Marwa-AL** [reply]
sorry for writing again. I thought this could be useful for identifying the issue. 
gml file comes with .gsf file, could it be that during conversion something is not linked which is causing the error.

**Reply by kenohori** [reply]
No problem, but you're in the wrong place for this discussion (cityjson specifications). If you're using citygml-tools, maybe try there? https://github.com/citygml4j/citygml-tools

**Reply by Marwa-AL** [reply]
Thank you for directing me to ask in citygml-tools. It is solved :)
